.docker/gui-docker

```diff
@@ -1,56 +1,139 @@
-#!/usr/bin/env bash
+#!/bin/bash -u
+
+# This script is used to run a docker container with graphics support.
+# http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration
 
-# This script is used to create and run a docker container with settings for gui interactions
 # All arguments to this script will be appended to a docker run command.
 # Example command line:
-# ./gui-docker -ti --rm moveit/moveit /bin/bash
-
-# XAUTH=/tmp/.docker.xauth
-# xauth nlist :0 | sed -e 's/^..../ffff/' | xauth -f $XAUTH nmerge -
-if [ ! -f /tmp/.docker.xauth ]
-then
-  export XAUTH=/tmp/.docker.xauth
-  xauth nlist :0 | sed -e 's/^..../ffff/' | xauth -f $XAUTH nmerge -
-fi
+# ./gui-docker -it --rm moveit/moveit:master-source /bin/bash
+
+function check_nvidia2() {
+    # If we don't have an NVIDIA graphics card, bail out
+    lspci | grep -qi "vga .*nvidia" || return 1
+    # If we don't have the nvidia runtime, bail out
+    if ! docker -D info | grep -qi "runtimes.* nvidia" ; then
+       echo "nvidia-docker v2 not installed (see https://github.com/NVIDIA/nvidia-docker/wiki)"
+       return 2
+    fi
+    echo "found nvidia-docker v2"
+    DOCKER_PARAMS="\
+      --runtime=nvidia \
+      --env=NVIDIA_VISIBLE_DEVICES=${NVIDIA_VISIBLE_DEVICES:-all} \
+      --env=NVIDIA_DRIVER_CAPABILITIES=${NVIDIA_DRIVER_CAPABILITIES:+$NVIDIA_DRIVER_CAPABILITIES,}all"
+    return 0
+}
+
+function check_nvidia1() {
+    # If we don't have an NVIDIA graphics card, bail out
+    lspci | grep -qi "vga .*nvidia" || return 1
+    # Check whether nvidia-docker is available
+    if ! which nvidia-docker > /dev/null ; then
+       echo "nvidia-docker v1 not installed either"
+       return 2
+    fi
+    # Check that nvidia-modprobe is installed
+    if ! which nvidia-modprobe > /dev/null ; then
+       echo "nvidia-docker-plugin requires nvidia-modprobe. Please install it!"
+       return 3
+    fi
+    # Retrieve device parameters from nvidia-docker-plugin
+    if ! DOCKER_PARAMS=$(curl -s http://localhost:3476/docker/cli) ; then
+       echo "nvidia-docker-plugin not responding on http://localhost:3476/docker/cli"
+       echo "See https://github.com/NVIDIA/nvidia-docker/wiki/Installation-(version-1.0)"
+       return 3
+    fi
+    echo "found nvidia-docker v1"
+    DOCKER_EXECUTABLE=nvidia-docker
+}
+
+function check_dri() {
+    # If there is no /dev/dri, bail out
+    test -d /dev/dri || return 1
+    DOCKER_PARAMS="--device=/dev/dri --group-add video"
+}
+
+function transfer_x11_permissions() {
+    # store X11 access rights in temp file to be passed into docker container
+    XAUTH=/tmp/.docker.xauth
+    touch $XAUTH
```

Comment:
```suggestion
    touch "$XAUTH"
```
I'll default to you, but my understanding of bash scripts is that all variables should be in quotes.

Reply:
Indeed, it's good practice to quote variables. In this case, it's not necessary, because the variable doesn't contain spaces.